### PR TITLE
[XENOARCH SPOILERS] [READY] Gigadrill

### PR DESCRIPTION
**Pull request by alexkar598**
# WARNING: SPOILERS FOR XENOARCH
___
# DO NOT SCROLL DOWN IF YOU WANT TO KEEP THE FUN OF DISCOVERING XENOARCH LOOT
.
.
.
.
.
.
.
.
.
.
.
.
.
.
.
.
.

Readds the giga drill back from 2013,improved by /vg/. It's a vehicle that you can ride to mine,it is faster than the ripley due to the fact it will be hard-ish to get via xenoarch.Currently only admin spawnnable,please do not add a way to get it other than xenoarch.Drag an drop ore box to it to load and drag and drop this to empty turf to unload.Mines and collects ore in a 1x3 range a la ripley.

No Change log due to this being unobtainable.

### CODE REVIEW

File: yogstation/code/modules/xenoarch/loot/gigadrill.dm [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+		icon_state = "gigadrill_mov"
+	else
+		icon_state = "gigadrill"
+
+/obj/vehicle/ridden/gigadrill/Initialize()
+	. = ..()
+	var/datum/component/riding/D = LoadComponent(/datum/component/riding)
+	D.vehicle_move_delay = 1
+	D.set_riding_offsets(RIDING_OFFSET_ALL, list(TEXT_NORTH = list(0, 7), TEXT_SOUTH = list(0, 18), TEXT_EAST = list(-18, 9), TEXT_WEST = list( 18, 9)))
+	D.set_vehicle_dir_layer(SOUTH, OBJ_LAYER)
```
Comment (alexkar598) [inline yogstation/code/modules/xenoarch/loot/gigadrill.dm]: Gotta change this to
```suggestion
	D.set_vehicle_dir_layer(SOUTH, ABOVE_MOB_LAYER)
```

File: yogstation/code/modules/xenoarch/loot/gigadrill.dm [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	name = "Giga Drill"
+	desc = "This looks like any normal drill,except not,it is a giga drill!"
+	icon = 'yogstation/icons/obj/xenoarch/artifacts.dmi'
+	icon_state = "gigadrill"
+	key_type = null //maybe add a key in the future
+
+	var/obj/structure/ore_box/OB
+
+/obj/vehicle/ridden/gigadrill/Destroy(force)
+	OB = null
```
Comment (Time-Green) [inline yogstation/code/modules/xenoarch/loot/gigadrill.dm]: Assuming the orebox is an in internal thing, QDEL_NULL(OB) is preferable

File: yogstation/code/game/gamemodes/traitor/double_agents.dm [hunk 3]
```diff
@@ -104,7 +104,7 @@
 			add_objective(martyr_objective)
 		else
 			var/datum/objective/hijack/hijack_objective = new
-			hijack_objective.owner = owner.
```
Comment (Altoids1) [inline yogstation/code/game/gamemodes/traitor/double_agents.dm:107]: What the hell...?

File: yogstation/code/modules/xenoarch/loot/gigadrill.dm [hunk 4]
```diff
@@ -0,0 +1,110 @@
+/obj/vehicle/ridden/gigadrill
+	name = "Giga Drill"
+	desc = "This looks like any normal drill,except not,it is a giga drill.Drag and drop an ore box on it to load the box on and drag and drop to a adjacant turf to unlload the box."
```
Comment (Altoids1) [inline yogstation/code/modules/xenoarch/loot/gigadrill.dm]: Put some spaces after your commas & periods, you cretin.

**Review by Altoids1 — CHANGES_REQUESTED** [review]

**Review by Altoids1 — APPROVED** [review]

**Comment by alexkar598** [comment]
Wrong button I hate my life and all form of dinosaurs.

**Pull request merged**